### ISSUE TITLE
Qt Message Handler null pointer dereference, and EUDC.TTE font issue

### ISSUE THREAD

**Issue by SimplexPL** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#130]
I have a weird issue, I stopped seeing OVRAS button in SteamVR Dashboard. I tried running it standalone, in Desktop mode, but it never starts (or it crashes immediately after starting). Please help me troubleshoot the issue - I can provide logs if I know where to find them.

(I also have fpsVR and it works)

**Comment by ykeara** [comment]
First of all I would try Update to Advanced settings 3.0.0 (just released).

But Logs (depending on version)

%APPDATA%\matzman666\ - pre 3.0.0
%APPDATA%\AdvancedSettings-Team\ - 3.0.0

**Comment by SimplexPL** [comment]
No changes with 3.0.0, still does not work. The log file is:
`[INFO] 2019-03-26 23:17:13: Application started (Version v3.0.0)
[INFO] 2019-03-26 23:17:13: Log File: C:/Users/Admin/AppData/Roaming/AdvancedSettings-Team/AdvancedSettings.log
[INFO] 2019-03-26 23:17:13: Command line arguments processed.
[INFO] 2019-03-26 23:17:13: OpenVR initialized successfully.
[INFO] 2019-03-26 23:17:13: Settings File: C:/Users/Admin/AppData/Roaming/AdvancedSettings-Team/OpenVRAdvancedSettings.ini
[INFO] 2019-03-26 23:17:13: VR Runtime Path: E:/Gry/Steam/steamapps/common/SteamVR
[INFO] 2019-03-26 23:17:13: Found overlay "valve.steam.desktop".`

**Comment by ykeara** [comment]
Did you use the installer? With steamvr running run advancedsettings.exe and restart then see if it works and the log if not

**Comment by SimplexPL** [comment]
Yes I did use the installer.
What should I restart?
I tried running the exe once again and the result is the same. It's not in the task manager and the log file is identical as above.

**Comment by SimplexPL** [comment]
Yes I did use the installer.
What should I restart?
I tried running the exe once again and the result is the same. It's not in the task manager and the log file is identical as above. The process crashes or is killed immediately after staring.

**Comment by ykeara** [comment]
steamVR, possibly your computer.... try running advancedsettings.exe with the launch argument "--force-install-manifest" and see if that fixes if not need to see log? are you using the new installer for 3.0.0?

**Comment by ykeara** [comment]
So it should show up under "vrcompositor" at least on Windows 10 have to expand it.

**Comment by SimplexPL** [comment]
I am using new 3.0.0 installer. I restarted PC, reinstalled OVRAS, I also tried using "--force-install-manifest".
Nothing changed, the process still crashes 1-2 seconds after starting. I found some crash dumps:
https://we.tl/t-Qr6LdU1slg

Log file: https://pastebin.com/raw/4Nmc4dZp

**Comment by MacMerc** [comment]
I had the same problem.  Advanced Settings icon appeared at bottom of SteamVR Overlay but window was invisible.  Discovered it was invisible because my controller laser was clearly hitting something.   Disconnected extra monitors and it showed up.
TLDR: Multi-monitor made OVRAS window disappear in my case.

**Comment by SimplexPL** [comment]
I only have one display and I don't even see the icon in SteamVR Overlay. The app crashes immediately after starting.

**Comment by ykeara** [comment]
@SimplexPL if you can post the logs for vrserver from steam (C:\Program Files (x86)\Steam\logs default) specifically vrserver.txt and vrserver.previous.txt

logs from OVRAS look good, but need to look elsewhere haven't had a chance to look ad .dmp's yet.

**Comment by username223** [comment]
Could you try downloading [this](http://s000.tinyupload.com/index.php?file_id=90271911083812185244) debug build, running it and then uploading the crash dumps?

**Comment by username223** [comment]
All previous crash dumps have a null pointer at `00007FF662FE28B3  cmp         byte ptr [rdx+r14],bl`.

`
Unhandled exception at 0x00007FF662FE28B3 (AdvancedSettings.exe) in AdvancedSettings.exe.100.dmp: 0xC0000005: Access violation reading location 0x0000000000000000.
`

**Comment by SimplexPL** [comment]
Sorry for delay in responses, I wil check the latest version.
Thanks for all your help.

**Comment by username223** [comment]
@SimplexPL 	When you have the time could you also list your OS, CPU, HMD, controllers, RAM and as much else as you can?

**Comment by username223** [comment]
Also, could you try running [this](https://uploadfiles.io/ociqx) file and uploading the crashdump?

**Comment by SimplexPL** [comment]
> [INFO] 2019-03-31 12:45:18: VR Runtime Path: E:/Gry/Steam/steamapps/common/SteamVR
[WARNING] 2019-03-31 12:45:18: Could not read "revive::ThumbSensitivity" setting: VRSettingsError_UnsetSettingHasNoDefault
[WARNING] 2019-03-31 12:45:18: Could not read "revive::User" setting: VRSettingsError_UnsetSettingHasNoDefault
[WARNING] 2019-03-31 12:45:18: Could not read "revive::Name" setting: VRSettingsError_UnsetSettingHasNoDefault
[INFO] 2019-03-31 12:45:21: Could not find overlay "valve.steam.desktop" after 7 extra attempts.

**Comment by username223** [comment]
@SimplexPL 
Which version is that log from?

Could you try running [this](https://uploadfiles.io/ac7ku) version and giving the full log?

Also, what are your specs?

**Comment by SimplexPL** [comment]
Log is from 3.0.1

My specs: Win 10 Pro, 1080Ti 419.67 drivers, SteamVR Beta, i7-9600K, Latest SteamVR beta.

I will try running the version you provided later and will let you know the results.

**Comment by SimplexPL** [comment]
This version works, thanks.
I noticed your comment in another issue:

> The Unable to open default EUDC font: "C:\\Windows\\FONTS\\EUDC.TTE" issue is on your end, so there's nothing we can do about it.

I have a very fresh win10 pro install and I have all the latest updates. I also checked my previous windows install where previous versions of OVRAS were working and that EUDC.TTE file is also not present. So I think it would be worth to investigate the origin of that issue.

**Comment by username223** [comment]
@SimplexPL
It's also not present for me on Win 10, but you two are the only people who have had problems with it.

Could you paste the logs of the working version?

**Comment by SimplexPL** [comment]
> [INFO] 2019-04-03 19:12:27: Application started (Version v3.0.0)
[INFO] 2019-04-03 19:12:27: Log File: C:/Users/Admin/AppData/Roaming/AdvancedSettings-Team/AdvancedSettings.log
[INFO] 2019-04-03 19:12:27: Command line arguments processed.
[INFO] 2019-04-03 19:12:28: OpenVR initialized successfully.
[INFO] 2019-04-03 19:12:28: Settings File: C:/Users/Admin/AppData/Roaming/AdvancedSettings-Team/OpenVRAdvancedSettings.ini
[INFO] 2019-04-03 19:12:28: VR Runtime Path: E:/Gry/Steam/steamapps/common/SteamVR
[WARNING] 2019-04-03 19:12:28: Could not read "revive::ThumbSensitivity" setting: VRSettingsError_UnsetSettingHasNoDefault
[WARNING] 2019-04-03 19:12:28: Could not read "revive::User" setting: VRSettingsError_UnsetSettingHasNoDefault
[WARNING] 2019-04-03 19:12:28: Could not read "revive::Name" setting: VRSettingsError_UnsetSettingHasNoDefault
[INFO] 2019-04-03 19:12:28: Found overlay "valve.steam.desktop".
[WARNING] 2019-04-03 19:12:29: Unable to open default EUDC font: "C:\\WINDOWS\\FONTS\\EUDC.TTE" (windows\qwindowsfontdatabase.cpp:1208)
[WARNING] 2019-04-03 19:12:35: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:12:41: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:12:47: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:12:52: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:12:58: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:03: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:09: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:14: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:20: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:26: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:31: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:37: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:42: Attempting to Reloading Chaperone Data
[WARNING] 2019-04-03 19:13:48: Attempting to Reloading Chaperone Data


What is that last warning? Why is it repeated so many time?

**Comment by username223** [comment]
@SimplexPL The last warning is when we're reading the chaperone data from the OpenVR API and we get invalid values. There's nothing we can do about it, it's entirely on Valve's end.

The crashing bug was the same as #139, this issue will be solved in the next release.

The EUDC font error could be because of language settings.

Which locale, language and keyboard layout are you using?

Does text render correctly in the overlay?
__
Comment:
`[WARNING] 2019-04-03 19:12:29: Unable to open default EUDC font: "C:\WINDOWS\FONTS\EUDC.TTE" (windows\qwindowsfontdatabase.cpp:1208)`
This method of error outputting did not work for the user in #139. The current method introduced in PR #144 will remain because it is unknown if `file` and `line` are supported on all systems.

**Comment by SimplexPL** [comment]
[https://i.imgur.com/GKhrD9j.png](url)

**Comment by username223** [comment]
(In case imgur link gets pulled: Language and keyboard is US English. Region and locale is Poland.)

There are no commonalities with #139 in setup besides Windows OS (7 in #139, 10 here) and running latin alphabets with additional characters.

#139 also did not have issues with rendering text in the overlay in spite of the font warning.

Since this issue has been fixed by PR #144 and the font warning does not seem to have any effect, this issue is considered resolved.

@SimplexPL Thanks for taking the time to create an issue, you helped resolve a pretty serious issue that would completely prevent people from running the application.

**Comment by SimplexPL** [comment]
I am glad I could help :) Thanks for picking up the mantle after matzman666 and continuing the development of OVRAS.
